js/socket.ts

```diff
@@ -0,0 +1,54 @@
+import { reload } from "./browser";
+import { getMetaContent } from "./util/metadata";
+import { Socket } from "phoenix";
+import { z } from "zod";
+
+export const initSocket = () => {
+  const clientRelease = getMetaContent("release");
+  const socket = new Socket("/socket", {
+    params: {
+      release: clientRelease,
+      token: getMetaContent("guardianToken"),
+    },
+  });
+
+  socket.onError((_error, _transport, _establishedConnections) => {
+    console.warn("socket error");
+  });
+
+  socket.onOpen(() => {
+    socket
+      .channel("metadata")
+      .join()
+      .receive("ok", (rawResponse: unknown) => {
+        const { authenticated, server_release: serverRelease } =
+          MetadataJoinResponse.parse(rawResponse);
```

Comment:
Would we prefer to catch zod parse errors here and log them to Sentry, but not throw a whole-page error and so on?

Reply:
This is already in an async handler, so if it fails it won't crash the whole app (and I just tested that). I'm not sure if that's a normal JS/browser thing or if it's because Phoenix has its own try/catch in `receive`.

Unfortunately, I don't think it gets sent to sentry. I'll look into that.